src/OpalCompiler-Core/PrimitiveErrorVariable.class.st

```diff
@@ -0,0 +1,35 @@
+"
+Primitives can declare a local variable that will hold the error code.
+"
+Class {
+	#name : #PrimitiveErrorVariable,
+	#superclass : #LocalVariable,
+	#instVars : [
+		'node'
+	],
+	#category : #'OpalCompiler-Core-Semantics'
+}
+
+{ #category : #'instance creation' }
+PrimitiveErrorVariable class >> node: aVariableNode [
+	^self new 
+		name: aVariableNode name;
+		node: aVariableNode;
+		markWrite;
+		yourself
+]
+
+{ #category : #visiting }
+PrimitiveErrorVariable >> acceptVisitor: aProgramNodeVisitor node: aNode [
+	^ aProgramNodeVisitor visitTemporaryNode: aNode
+]
+
+{ #category : #queries }
+PrimitiveErrorVariable >> definingNode [
+	^node
```

Comment:
I think it would be better to call this variable #definingNode. 
I know it was an existing logic to use a variable node as result. But I wonder why it done like that?
I would expect a pragma node to be a definition. Or even relax to method node as default for other temps (which will simplify the implementation).